### ISSUE TITLE
SPFX pnp sp get list item failing in Edge and IE11.

### ISSUE THREAD

**Issue by paigeflourin** [pnp/pnpjs#356]
#### Category
- [ ] Question
- [ ] Typo
- [x ] Bug
- [ ] Additional article idea

I am having issues with running my SPFX app on Edge. I have a functionality that gets a certain list item and check if it is locked by a user. this is my code:

```
 public lockList (currentUser: any, division: string, qnaListTrackingListName: string) : Promise<any>{
        //let res; 
        let d = new Date();
        return sp.web.lists.getByTitle(qnaListTrackingListName).items.top(1).filter("Division eq '" + division + "'").get().then((items: any[]) => {
            // see if we got something
            if (items.length > 0) {
                return  sp.web.lists.getByTitle(qnaListTrackingListName).items.getById(items[0].Id).update({
                    LockedById: currentUser.Id,
                    LastUpdated: d.toLocaleDateString()
                }).then(result => {
                    console.log(result);
                    return result;
                }).catch(error => {
                    console.log();
                    return error;
                });
            }
        });
```

#### Expected or Desired Behavior

this should check if the item is currently locked by another user. 

#### Observed Behavior

It gets the list item i have logged it in console but it then have the error once i updated the item that i have retrieved. 

it is working in Chrome perfectly but it is not working in Edge i get this error in Edge:

>  HTTP400: BAD REQUEST - The request could not be processed by the server due to invalid syntax.
> (Fetch)POST - https://siteURL/_api/web/lists/getByTitle('QnA List Tracking')/items(1)

#### Steps to Reproduce

https://github.com/paigeflourin/test

* i would have to email you the configuration for the spfx webpart in order to run this one.

**Comment by koltyakov** [comment]
@paigeflourin,

Did you install polyfills? More details: https://pnp.github.io/pnpjs/documentation/polyfill/

**Comment by paigeflourin** [comment]
hi @koltyakov ,

i did but not sure if am doing it right have installed and imported: 

require('core-js');
import 'es6-promise';
require('es6-promise/auto');
import 'whatwg-fetch';


also included this before pnp/sp declaration: 

import "@pnp/polyfill-ie11";

**Comment by paigeflourin** [comment]
@koltyakov  if this is helpful, this is my tsconfig:
```

{
  "compilerOptions": {
    "target": "es5",
    "forceConsistentCasingInFileNames": true,
    "module": "esnext",
    "moduleResolution": "node",
    "jsx": "react",
    "declaration": true,
    "sourceMap": true,
    "experimentalDecorators": true,
    "skipLibCheck": true,
    "outDir": "lib",
    "typeRoots": [
      "./node_modules/@types",
      "./node_modules/@microsoft"
    ],
    "types": [
      "es6-promise",
      "webpack-env"
    ],
    "lib": [
      "es5",
      "es6",
      "dom",
      "es2015.collection",
      "es2015.promise",
      "es2016"
    ]
  },
  "include": [
    "src/**/*.ts"
  ],
  "exclude": [
    "node_modules",
    "lib"
  ]
}
```

this is also the package dependencies I have installed that might relate to polyfill and fetch

"@babel/polyfill": "^7.0.0",
 "@microsoft/sp-polyfills": "^1.7.0",
 "@pnp/polyfill-ie11": "^1.0.0",
    "@pnp/sp": "^1.2.1",
    "@types/es6-promise": "0.0.33",
 "@types/webpack-env": "1.13.1",
    "@types/whatwg-fetch": "0.0.33",
"es6-promise": "^4.2.5",
 "whatwg-fetch": "^3.0.0"

**Comment by koltyakov** [comment]
So it looks like you never install `@pnp/sp`'s peer dependencies?
Please check the docs installation article: https://pnp.github.io/pnpjs/documentation/deployment/#install_1
TL;DR: `npm install @pnp/logging @pnp/common @pnp/odata @pnp/sp`. Also, as peer dependencies, please make sure that all the `@pnp/*` dependencies have the same version.

**Comment by paigeflourin** [comment]
@koltyakov 

i did

  "@pnp/common": "^1.2.1",
    "@pnp/graph": "^1.2.1",
    "@pnp/logging": "^1.2.1",
    "@pnp/odata": "^1.2.1",
    "@pnp/polyfill-ie11": "^1.0.0",
    "@pnp/sp": "^1.2.1",

**Comment by patrick-rodgers** [comment]
Is there a reason you are using odata verbose? Could you try removing that in the setup and see is things work, then at least that would give us a hint at where to look.

**Comment by paigeflourin** [comment]
Hi @patrick-rodgers  tried removing the odata verbose is till get my edge error :(

**Comment by paigeflourin** [comment]
HI @patrick-rodgers  but the good news is that i can now see it in IE! 

same error as that of Edge, but i saw a more helpful error. just a question though. what date tiem syntax does Edge and IE accept? 

![image](https://user-images.githubusercontent.com/6951524/48295609-7c9c2480-e4c8-11e8-9848-6ad591cbcaad.png)

I am using this in my code:

```
sp.web.lists.getByTitle(qnaListTrackingListName).items.getById(items[0].Id).update({
                    LockedById: currentUser.Id,
                    LastUpdated: d.toLocaleDateString()
                })
```

**Comment by paigeflourin** [comment]
@koltyakov @patrick-rodgers , managed to get it working by using moment js instead! THANK YOU!
